pkg/cloud/services/compute/instance.go

```diff
@@ -596,61 +530,148 @@ func (s *Service) DeleteInstance(object runtime.Object, instanceName string) err
 	}
 	// get and delete trunks
 	for _, port := range instanceInterfaces {
-		err := attachinterfaces.Delete(s.computeClient, instance.ID, port.PortID).ExtractErr()
-		if err != nil {
+		if err = deleteAttachinterface(s, object, instance.ID, port.PortID); err != nil {
 			return err
 		}
+
 		if trunkSupport {
-			listOpts := trunks.ListOpts{
-				PortID: port.PortID,
-			}
-			allTrunks, err := trunks.List(s.networkClient, listOpts).AllPages()
-			if err != nil {
-				return err
-			}
-			trunkInfo, err := trunks.ExtractTrunks(allTrunks)
-			if err != nil {
+			if err = deleteTrunk(s, object, port.PortID); err != nil {
 				return err
 			}
-			if len(trunkInfo) == 1 {
-				err = util.PollImmediate(RetryIntervalTrunkDelete, TimeoutTrunkDelete, func() (bool, error) {
-					if err := trunks.Delete(s.networkClient, trunkInfo[0].ID).ExtractErr(); err != nil {
-						if capoerrors.IsRetryable(err) {
-							return false, nil
-						}
-						return false, err
-					}
-					return true, nil
-				})
-				if err != nil {
-					return fmt.Errorf("error deleting the trunk %v", trunkInfo[0].ID)
-				}
+		}
+
+		if err = deletePort(s, object, port.PortID); err != nil {
+			return err
+		}
+	}
+
+	return deleteInstance(s, object, instance.ID)
+}
+
+func deletePort(is *Service, object runtime.Object, portID string) error {
+	port, err := ports.Get(is.networkClient, portID).Extract()
+	if err != nil {
+		if capoerrors.IsNotFound(err) {
+			return nil
+		}
+	}
```

Comment:
Shouldn't the error still get returned if it is not `IsNotFound`?
```suggestion
		return err
	}
```